doc/source/ray-core/tasks/scheduling.rst

```diff
@@ -22,11 +22,28 @@ See :ref:`Placement Group <ray-placement-group-doc-ref>` for more details.
 Scheduling Strategy
 -------------------
 Tasks support ``scheduling_strategy`` option to specify the strategy used to decide the best node among available nodes.
-Currently the supported strategies are "DEFAULT" and "SPREAD".
+Currently the supported strategies are ``"DEFAULT"``, ``"SPREAD"`` and
+``ray.util.scheduling_strategies.NodeSchedulingStrategy(node_id, soft)``.
+
 "DEFAULT" is the default strategy used by Ray. With the current implementation, Ray will try to pack tasks on nodes
 until the resource utilization is beyond a certain threshold and spread tasks afterwards.
+
 "SPREAD" strategy will try to spread the tasks among available nodes.
 
+NodeSchedulingStrategy is the low level strategy that allows a task to be scheduled onto a particular node specified by the hex node id.
+The ``soft`` flag specifies whether the task is allowed to run somewhere else if the specified node doesn't exist (e.g. the node dies)
+or is infeasible for the required resources. If it is True, the task will be scheduled onto other feasible nodes under such case.
+If it's False, the task will fail with ``TaskUnschedulableError``.
+As long as the specified node is alive and feasible, the task will only run there
+regardless of the ``soft`` flag. This means if the node currently has no available resources, the task will wait until resources
+become available.
+This strategy should ONLY be used if other high level scheduling strategies (e.g. placement group) cannot give the
+desired task placements. It has the following known limitations:
+1. It's a low level strategy which prevents optimizations by a smart scheduler.
+2. It cannot fully utilize an autoscaling cluster since node ids must be known when the tasks are created.
+3. Users normally don't have enough information to make the best static task placement decision
+especially in a multi-tenant cluster: for example, they don't know what else are being scheduled onto those nodes as well.
```

Comment:
```suggestion
3. It can be difficult to make the best static task placement decision
especially in a multi-tenant cluster: for example, an application won't know what else is being scheduled onto the same nodes.
```